core/domain/wipeout_service.py

```diff
@@ -685,11 +741,17 @@ def _collect_and_save_entity_ids_from_snapshots_and_commits(
         for model in snapshot_metadata_models)
 
     commit_log_ids = set()
-    commit_log_models = []
+    commit_log_models: List[base_models.BaseCommitLogEntryModel] = []
     if commit_log_model_class is not None:
-        commit_log_models = commit_log_model_class.query(
+        commit_log_models = list(commit_log_model_class.query(
             commit_log_model_class.user_id == user_id
-        ).fetch()
+        ).fetch())
+        # Here `commit_log_model_field_name` is never going to be a None
+        # value, because commit_log_model_field_name can only be None when
+        # commit_log_model_class is None and if commit_log_model_class is
+        # None, then this line of code is never going to be executed. So,
+        # to just narrow down type for type checking, we used assert here.
+        assert commit_log_model_field_name is not None
```

Comment:
I think you should have something like: if commit log model class exists but the field name is not specified, throw error.

Try not to rely on runtime asserts, where possible.

Reply:
Done, an exception is added docstring is also updated!